OTAnalytics/plugin_prototypes/track_visualization/track_viz.py

```diff
@@ -283,18 +295,85 @@ def _convert_tracks(self, tracks: Iterable[Track]) -> DataFrame:
             DataFrame: a dataframe containing the detections of the given tracks.
         """
         if self._cache_df.empty:
-            self._cache_df = super()._convert_tracks(tracks)
+            self._cache_df = self.__do_convert_tracks(tracks)
 
         return self._cache_df
 
+    def __do_convert_tracks(self, tracks: Iterable[Track]) -> DataFrame:
+        """Internal method to convert the given tracks to dataframe format without
+        caching.
+
+        Args:
+            tracks (Iterable[Track]): the tracks to be converted.
+
+        Returns:
+            DataFrame: the dataframe representation of the given tracks.
+        """
+        return super()._convert_tracks(tracks)
+
     def notify_tracks(self, tracks: list[TrackId]) -> None:
         """Take notice of some change in the track repository.
-        Resets the cached dataframe, so it will be recomputed.
+
+        Remove cached tracks matching any given id.
```

Comment:
Tracks with `TrackId`s that already exist in the cache should not be simply discarded. 

A `Track` with an unique `TrackId` can be continuous and thus appear in more than one ottrk file.
In that case the ottrk parser will take the existing `Track` and merge it with the new `Track` with the same id and update the track repository.

Thus an already existing TrackId being passed to the `notify_tracks` method indicates that there was actually an update of the respective track. In this case the correct behaviour would be to remove the track in the cache and replace it with the updated one.

In the future there are plans to have separate observers for removing and adding to a repository.